src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -381,15 +381,32 @@ def schedule_unit(self, unit):
         is_last  = False
         tag      = cud.get('tag')
 
+        # - PRRTE related - start -
+        dvm_hosts_list = self._rm_lm_info.get('dvm_hosts') or []
+        # use tag as a DVM ID, so task will be allocated to nodes that are
+        # handled by a corresponding DVM (only applied with `dvm_hosts_list`)
+        # FIXME: new coming attribute `tags` would consider `dvm_id`
+        if dvm_hosts_list and tag is not None and tag not in self._tag_history:
```

Comment:
Why is the tag converted to int only when it's not in the tag history?  Shouldn't it *always* be converted?  I also would suggest to move the tag_history entry to lower, so that we have a single place where the history is changed. (see comment there). 

I would suggest for here:
```py
if dvm_host_list and tag:
    try:
        tag = int(tag)
        if len(dvm_hosts_list) <= tag:
                raise ValueError('dvm_id (%s) out of range' % tag)
    except (TypeError, ValueError):
        tag = None
````

Reply:
> Why is the tag converted to int only when it's not in the tag history? Shouldn't it always be converted?

oh, absolutely valid, somehow missed that, yes, should be converted before being checked

and couple responses on other points:
- if `tag` is set and we have non-empty `dvm_host_list`, which means that DVMs will be set, then `tag` is considered not as a label for host(s) but as an ID for DVM. And tag should be added to the `self._tag_history` before going through the list of available hosts. I would think that after it will migrate to `tags` it will be with a valid name `dvm_id`;
- speaking about your example: I did separate tag/dvm_id value as if it is non-numeric then we just ignore it, but if it is a numeric value then it should be within a range of DVM IDs, thus if this `len(dvm_hosts_list) <= tag` is valid then exception should be raised

Reply:
The tags PR is close to being merged, too, so that all will soon cleanup anyway.  Thanks for clarifying! :-)